Framework/PythonInterface/test/python/mantid/geometry/ComponentInfoTest.py

```diff
@@ -105,5 +106,112 @@ def test_name(self):
         info = self._ws.componentInfo()
         self.assertEquals(type(info.name(0)), str)
 
+
+    """ 
+    The section below is for destructive tests - i.e extreme
+    and exceptional test cases. 
+    """
+
+    def test_detectorsInSubtree_exceptional(self):
+        info = self._ws.componentInfo()
+        with self.assertRaises(TypeError):
+        	info.detectorsInSubtree("Error")
+        with self.assertRaises(TypeError):
+        	info.detectorsInSubtree(10.0)
+        with self.assertRaises(Exception):
+        	info.detectorsInSubtree(100)
```

Comment:
Not clear why this would fail. Maybe that should be in it's own test with the method name describing (be verbose) what is expected.